bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -848,6 +850,21 @@ protected void runRule(RuntimeRule rule, RuleEngineCallbackImpl.TriggerData td)
 
     }
 
+    private void clearCacheOutputValues(RuntimeRule rule) {
+        List<Trigger> tiggers = rule.getTriggers();
+        for (Trigger trigger : tiggers) {
+            ((RuntimeTrigger) trigger).setOutputs(null);
```

Comment:
> but we have to clean A's ouputs from previous execution

Why should you? As long as you only put the output of executed modules into the context, you don't have to worry about any other (inactive) module...